scratch/copy-mk.py

```diff
@@ -0,0 +1,155 @@
+    parser.add_argument("--cmc-port", type=int, default=7147, help="KATCP port of the CMC. [%(default)s]")
+    parser.add_argument("--subordinate", type=str, required=True, help="Which subordinate to copy.")
+    parser.add_argument(
+        "--develop",
+        nargs="?",
+        const=True,
+        help="Pass development options in the config. Use comma separation, or omit the arg to enable all.",
+    )
+    parser.add_argument("--image-override", action="append", metavar="NAME:IMAGE:TAG", help="Override a single image")
+    args = parser.parse_args(argv)
+    return args
+
+
+async def async_main(args) -> int:
+    cmc_client = await aiokatcp.Client.connect(args.cmc_host, args.cmc_port)
+
+    try:
+        _, informs = await cmc_client.request("subordinate-list")
+    except (aiokatcp.FailReply, ConnectionError) as exc:
+        print(f"Error: {exc}", file=sys.stderr)
+        return 1
+
+    if len(informs) == 0:
+        print(f"No subarrays currently running on {args.cmc_host}")
+        return 1
+
+    subordinates = {}
+    for inform in informs:
+        name = inform.arguments[0].decode()
+        control_port, monitor_port = inform.arguments[1].decode().split(",")
+        control_port = int(control_port)
+        monitor_port = int(monitor_port)
+        mcast_groups = []
+        for grp in inform.arguments[2:]:
+            mcast_groups.append(grp.decode())
+        n_antennas = len(mcast_groups) // 2
+        for n, grp in enumerate(mcast_groups[1:]):
+            if grp == mcast_groups[0]:
+                # MK is not using a full power-of-2 size array, there are dummies from here onwards.
+                print("Dummy antenna found starting at ", (n + 1) // 2)
+                n_antennas = (n + 1) // 2
+                break
+        digitiser_address = mcast_groups[0].split("+")[0]
+
+        subordinates[name] = Subordinate(name, control_port, monitor_port, digitiser_address, n_antennas)
+
+    try:
+        target = subordinates[args.subordinate]
+    except KeyError:
+        print(f"{args.subordinate} is not among the subordinates currently running. Options are: {subordinates.keys()}")
+        return 1
+
+    # connect to the corr2_servlet itself to see the sync time, n_chans and bandwidth
+    corr2_client = await aiokatcp.Client.connect(args.cmc_host, target.control_port)
+    sync_time = await corr2_client.sensor_value("sync-time", float)
+    channels = await corr2_client.sensor_value("antenna-channelised-voltage-n-chans", int)
+    sample_rate = await corr2_client.sensor_value("antenna-channelised-voltage-bandwidth", float) * 2
+
+    # This won't distinguish between the different kinds of S-band. It'll be wrong. I'm not quite
+    # sure at this point how to match up the centre_frequency values in katgpucbf.meerkat.BANDS with
+    # what MK actually reports, even in the L and UHF cases, so for the purposes of getting
+    # something producing a result, I am leaving it as-is for the time being.
```

Comment:
I wouldn't worry about it too much for now - I'm not actually sure why we put band and centre frequency into the `dig.baseband_voltage` stream type, since we don't use them. Perhaps at the time I thought CBF could take over some of the delay manager functionality, which would require it. It's also needed if we're running Pipelines in the same subarray product (without CAM), or if we wanted to make fancier dsims that model geometric delays.

We might want to make those fields optional, at least for `dig.baseband_voltage`.